src/app/member/components/MemberFormBuilder.tsx

```diff
@@ -0,0 +1,396 @@
+"use client";
+import React from "react";
+import { useMask } from "@react-input/mask";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+import { addDays, format } from "date-fns";
+import { Shift, UserRole } from "@/constant/constant";
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { cn } from "@/lib/utils";
+import { CalendarIcon } from "lucide-react";
+import { Calendar } from "@/components/ui/calendar";
+import FormWrapper from "../../../common/FormWrapper";
+import { Label } from "@/components/ui/label";
+import { Textarea } from "@/components/ui/textarea";
+import {
+  CNIC_VALIDATOR_REGEX,
+  PHONENUMBER_VALIDATOR_REGEX,
+} from "../../../../regex";
+import { createUser } from "../actions/action";
+import { useToast } from "@/hooks/use-toast";
+
+export type UserBuilder = z.infer<typeof userBuilderSchema>;
+
+const USER_ROLES = [
+  {
+    role: "Member",
+    value: UserRole.Member,
+  },
+
+  {
+    role: "Shift Incharge",
+    value: UserRole.ShiftIncharge,
+  },
+  {
+    role: "Incharge",
+    value: UserRole.Incharge,
+  },
+];
+
+const SHIFTTIMING = [
+  {
+    time: "Shift 12:00am to 00:06am",
+    value: Shift.ShiftA,
+  },
+  {
+    time: "Shift 00:06am to 00:12pm",
+    value: Shift.ShiftB,
+  },
+  {
+    time: "Shift 00:12pm to 00:06pm",
+    value: Shift.ShiftC,
+  },
+  {
+    time: "Shift 00:06pm to 00:12am",
+    value: Shift.ShiftD,
+  },
+];
+
+export const userBuilderSchema = z.object({
+  name: z.string().min(1, {
+    message: "Name is required",
+  }),
+  phoneNumber: z
+    .string({ message: "Phone is required" })
+    .regex(PHONENUMBER_VALIDATOR_REGEX, "Phone number is not valid"),
+  date_of_birth: z.date().max(new Date(Date.now()), "under age"),
+  cnic: z.string().regex(CNIC_VALIDATOR_REGEX, "CNIC is not valid"),
+  shift: z.enum(["A", "B", "C", "D"], {
```

Comment:
This should be `shift: z.enum([Shift.ShiftA, Shfit.ShiftB...]`

Reply:
fixed